src/shared/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react'
+import Icon from '../Icon'
+import { Checkmark, Container, InnerContainer, Input } from './Checkbox.styles'
+
+type HTMLCheckboxProps = React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>
+export interface CheckboxProps extends Omit<HTMLCheckboxProps, 'value' | 'onChange' | 'checked' | 'multiple' | 'ref'> {
+  value: boolean
+  disabled?: boolean
+  indeterminate?: boolean
+  error?: boolean
+  className?: string
+  onChange?: (value: boolean) => void
+  onFocus?: (event: React.FocusEvent<HTMLDivElement>) => void
+  onBlur?: (event: React.FocusEvent<HTMLDivElement>) => void
+}
+
+export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
+  ({ value, indeterminate, onChange, disabled = false, onFocus, onBlur, error = false, className, ...props }, ref) => {
+    const isIndeterminate = indeterminate || false
+    const isSelected = !!value
+    const [isHovered, setIsHovered] = useState(false)
+    const [isFocused, setIsFocused] = useState(false)
+
+    const onMouseEnter = () => {
+      if (!disabled) {
+        setIsHovered(true)
+      }
+    }
+    const onMouseLeave = () => {
+      if (!disabled) {
+        setIsHovered(false)
+      }
+    }
+    const onChangeHandler = (e: React.FocusEvent<HTMLInputElement>) => {
+      if (!disabled && onChange) {
+        onChange(!value)
+      }
+    }
+    const onFocusHandler = (e: React.FocusEvent<HTMLInputElement>) => {
+      if (!disabled) {
+        setIsFocused(true)
+        if (onFocus) {
+          onFocus(e)
+        }
+      }
+    }
+    const onBlurHandler = (e: React.FocusEvent<HTMLInputElement>) => {
+      if (!disabled) {
+        setIsFocused(false)
+        if (onBlur) {
+          onBlur(e)
+        }
+      }
+    }
+    return (
+      <Container
+        onMouseEnter={onMouseEnter}
```

Comment:
Still think it's much better to remove `onMouseEnter` and `onMouseLeave`. You can just update your `Container` like this
```js
export const Container = styled.div<CheckboxStateProps>`
  position: relative;
  width: max-content;
  padding: ${sizes(2)};
  margin: -${sizes(2)};
  cursor: pointer;
  border-radius: 100%;
  color: ${colors.gray[300]};
  transition: background ${transitions.timings.loading} ${transitions.easing};
  ${({ isFocused, selected, disabled }) =>
    isFocused &&
    `background: ${colors.gray[800]};
    ${selected && `background: ${colors.blue[900]};`}
    ${disabled && `background: transparent;`}`}

  ${({ disabled }) =>
    !disabled &&
    css`
      :hover {
        background: ${colors.gray[800]};
      }
    `};
`
```
And of course, remove `onMouseEnter` and `onMouseLeave` handlers.

Reply:
+1, I totally agree, let's try to do that with just CSS

We could also get rid of focus/blur handlers if we used `:focus-within`, but the support isn't great so let's postpone that for now

Reply:
Aight, did pretty much that but with a tweak